File: lib/java/src/main/java/io/github/brenbar/uapi/internal/generation/GenerateRandomBoolean.java
package io.github.brenbar.uapi.internal.generation;

import io.github.brenbar.uapi.RandomGenerator;

public class GenerateRandomBoolean {
    public static Object generateRandomBoolean(Object blueprintValue, boolean useBlueprintValue,
            RandomGenerator randomGenerator) {
        if (useBlueprintValue) {
            return blueprintValue;
        } else {
            return randomGenerator.nextBoolean();
        }
    }
}

File: lib/ts/uapi/internal/generation/GenerateRandomBoolean.ts
import { RandomGenerator } from 'uapi/RandomGenerator';

export function generateRandomBoolean(
    blueprintValue: any,
    useBlueprintValue: boolean,
    randomGenerator: RandomGenerator,
): any {
    if (useBlueprintValue) {
        return blueprintValue;
    } else {
        return randomGenerator.nextBoolean();
    }
}
